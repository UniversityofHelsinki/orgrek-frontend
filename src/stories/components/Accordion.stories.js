import React from 'react';
import {
  Accordion,
  AccordionSummary,
  AccordionDetails,
} from '../../components/Accordion';

export default {
  component: Accordion,
  parameters: {
    docs: {
      description: {
        component: 'MUI Accordion customized with HY Design System styles',
      },
    },
  },
  argTypes: {
    onChange: { action: true, control: 'none' },
  },
};

export const Collapsed = {
  args: {
    disabled: false,
  },
  render: (args) => {
    return (
      <Accordion {...args}>
        <AccordionSummary>Accordion title</AccordionSummary>
        <AccordionDetails>Accordion details</AccordionDetails>
      </Accordion>
    );
  },
};

export const Expanded = {
  ...Collapsed,
  args: {
    ...Collapsed.args,
    defaultExpanded: true,
  },
};

export const Multiple = {
  render: () => (
    <div>
      <Accordion>
        <AccordionSummary>Accordion title 1</AccordionSummary>
        <AccordionDetails>Accordion details</AccordionDetails>
      </Accordion>
      <Accordion>
        <AccordionSummary>Accordion title 2</AccordionSummary>
        <AccordionDetails>Accordion details</AccordionDetails>
      </Accordion>
      <Accordion defaultExpanded>
        <AccordionSummary>Accordion title 3</AccordionSummary>
        <AccordionDetails>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras dapibus
          vulputate diam eu pretium. Mauris elit orci, ultricies id fermentum
          vel, porta et eros. Vestibulum condimentum lectus in convallis
          feugiat. Sed vulputate fringilla felis. Aliquam ut arcu et dui feugiat
          scelerisque eu quis diam. Mauris placerat congue dui sit amet blandit.
          Phasellus condimentum libero vel velit auctor, sit amet tincidunt
          velit varius.
        </AccordionDetails>
      </Accordion>
      <Accordion>
        <AccordionSummary>
          An example of an extremely long accordion title that contains
          uncomfortably many complicated words
        </AccordionSummary>
        <AccordionDetails>Accordion details</AccordionDetails>
      </Accordion>
    </div>
  ),
};
