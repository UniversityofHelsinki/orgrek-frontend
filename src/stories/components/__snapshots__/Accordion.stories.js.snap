// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stories/components/Accordion Collapsed smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Accordion title",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
  ],
  "name": "stories-components-accordion--collapsed",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;

exports[`stories/components/Accordion Expanded smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "expanded": true,
      "name": "Accordion title",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Accordion details",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
  ],
  "name": "stories-components-accordion--expanded",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;

exports[`stories/components/Accordion Multiple smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Accordion title 1",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Accordion title 2",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "expanded": true,
      "name": "Accordion title 3",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras dapibus vulputate diam eu pretium. Mauris elit orci, ultricies id fermentum vel, porta et eros. Vestibulum condimentum lectus in convallis feugiat. Sed vulputate fringilla felis. Aliquam ut arcu et dui feugiat scelerisque eu quis diam. Mauris placerat congue dui sit amet blandit. Phasellus condimentum libero vel velit auctor, sit amet tincidunt velit varius.",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "An example of an extremely long accordion title that contains uncomfortably many complicated words",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
  ],
  "name": "stories-components-accordion--multiple",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;
